packages/api-client-core/src/InternalModelManager.ts

```diff
@@ -212,7 +235,11 @@ export type RecordData = Record<string, any>;
 export class InternalModelManager {
   private readonly capitalizedApiIdentifier: string;
 
-  constructor(private readonly apiIdentifier: string, readonly connection: GadgetConnection) {
+  constructor(
+    private readonly apiIdentifier: string,
+    readonly connection: GadgetConnection,
+    readonly options: { pluralApiIdentifier: string }
```

Comment:
Could we make this optional and check in `bulkCreate` if it's set? Then you can throw an error (maybe `GadgetClientError`) when calling `bulkCreate` without a `pluralApiIdentifier`.